### PR TITLE
[InstCombine] Don't look at ConstantData users

### PR DESCRIPTION
**Pull request by aengelke**
When looking at PHI operand for combining, only look at instructions and
arguments. The loop later iteraters over Arg's users, which is not
useful if Arg is a constant -- it's users are not meaningful and might
be in different functions, which causes problems for the dominates()
query.

### CODE REVIEW

File: llvm/test/Transforms/InstCombine/phi-int-users.ll [hunk 1]
```diff
@@ -0,0 +1,576 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -mtriple=arm64 -passes='inline,function(sroa<modify-cfg>,jump-threading,instcombine)' -S < %s | FileCheck %s
```
Comment (nikic) [inline llvm/test/Transforms/InstCombine/phi-int-users.ll:2]: Why can't this be an instcombine only test?

Comment (aengelke) [reply llvm/test/Transforms/InstCombine/phi-int-users.ll]: True, changed

File: llvm/test/Transforms/InstCombine/phi-int-users.ll
Comment (nikic) [inline llvm/test/Transforms/InstCombine/phi-int-users.ll:1]: This test still doesn't look anything approaching minimal. Please run it through llvm-reduce.

File: llvm/test/Transforms/InstCombine/phi-int-users.ll [hunk 2]
```diff
@@ -0,0 +1,918 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -mtriple=arm64 -S < %s -passes=instcombine | FileCheck %s
```
Comment (nikic) [inline llvm/test/Transforms/InstCombine/phi-int-users.ll:2]: Triple should not be needed.

**Review by nikic — APPROVED** [review]
LGTM

**Review by fhahn — APPROVED** [review]
LGTM, thanks for the fix

**Pull request merged**